File: ROUND_1/CPP/Brute_force/Production_line_recovery.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int n;
    cin >> n;

    vector<int> arr(n);
    for (int i = 0; i < n; i++)
        cin >> arr[i];

    vector<int> sorted = arr;
    sort(sorted.begin(), sorted.end());

    // Compare against the fully sorted array to find the first
    // and last position that differ
    int left = 0;
    while (left < n && arr[left] == sorted[left])
        left++;

    // Already sorted
    if (left == n) {
        cout << 0 << endl;
        return 0;
    }

    int right = n - 1;
    while (arr[right] == sorted[right])
        right--;

    cout << right - left + 1 << endl;

    return 0;
}
/*
Time Complexity: O(n log n)
Space Complexity: O(n)
*/

File: ROUND_1/JAVA/Brute_force/Production_line_recovery.java

import java.io.*;
import java.util.*;


public class Production_line_recovery {
     static boolean isSorted(int[] arr) {

        for (int i = 1; i < arr.length; i++) {
            if (arr[i] < arr[i - 1])
                return false;
        }

        return true;
    }

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();

        int[] arr = new int[n];

        for (int i = 0; i < n; i++)
            arr[i] = sc.nextInt();

        if (isSorted(arr)) {
            System.out.println(0);
            return;
        }

        int answer = n;

        for (int left = 0; left < n; left++) {

            for (int right = left; right < n; right++) {

                int[] temp = arr.clone();

                Arrays.sort(temp, left, right + 1);

                if (isSorted(temp)) {
                    answer = Math.min(answer, right - left + 1);
                }
            }
        }

        System.out.println(answer);
    }
    
}
/* 
Time Complexity: O(n^3)
Space Complexity: O(n)
*/
